### PR TITLE
Entity manager refactoring

### PR DESCRIPTION
**Pull request by bohdan-harniuk**
**Description** (*)
I've refactored the entity manager feature, generators and related files in source code and tests.

**Contribution checklist** (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with integration/functional tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: src/com/magento/idea/magento2plugin/actions/generation/data/dialog/EntityManagerContextData.java [hunk 1]
```diff
@@ -0,0 +1,381 @@
+/*
```
Comment (VitaliyBoyko) [inline src/com/magento/idea/magento2plugin/actions/generation/data/dialog/EntityManagerContextData.java:1]: Let's call it Entity Creator. The  Entity manager, it's something different in Magento 2. It was just an initial idea. Let's rename it.

**Review by VitaliyBoyko — APPROVED** [review]
Okay, let’s fix the naming issue in the next PR. Too much effort as for a single one.

File: src/com/magento/idea/magento2plugin/actions/generation/data/dialog/EntityManagerContextData.java (hunk 1 above)
Comment (VitaliyBoyko) [reply src/com/magento/idea/magento2plugin/actions/generation/data/dialog/EntityManagerContextData.java:1]: This feature gonna be a game-changer. Great job!

Comment (bohdan-harniuk) [reply src/com/magento/idea/magento2plugin/actions/generation/data/dialog/EntityManagerContextData.java:1]: Thank you! I will do that in the next PR!

**Pull request merged**